src/components/directory-item-selector.js

```diff
@@ -123,14 +127,31 @@ const DirectoryItemSelector = (props) => {
     const fetchDirectory = useCallback(
         (nodeId) => {
             fetchDirectoryContent(nodeId)
-                .then((childrenToBeInserted) => {
-                    // update directory Content
-                    addToDirectory(
-                        nodeId,
-                        childrenToBeInserted.filter((item) =>
-                            contentFilter().has(item.type)
-                        )
+                .then((children) => {
+                    const childrenMatchedTypes = children.filter((item) =>
+                        contentFilter().has(item.type)
                     );
+                    if (
+                        props.equipmentTypes &&
+                        props.equipmentTypes.length > 0
+                    ) {
+                        // filtering also with equipment types
+                        fetchElementsMetadata(
```

Comment:
is it possible to filter elements from backend instead of fetching everything and filter them here ?